TestPointQueries.testAllPointDocsWereDeletedAndThenMergedAgain reproducible test failure

### Description

Happened in [this CI build](https://ci-builds.apache.org/job/Lucene/job/Lucene-Check-main/10750/) and also when I ran smoke tester for 9.9.0 RC 0.  It repros for me:

```
Stack Trace:
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=981, name=Lucene Merge Thread #2, state=RUNNABLE, group=TGRP-TestPointQueries]
        at __randomizedtesting.SeedInfo.seed([F9345050587589D4:3ED723D709FA79D1]:0)
Caused by: org.apache.lucene.index.MergePolicy$MergeException: java.lang.NullPointerException: Cannot invoke "org.apache.lucene.index.PointValues.size()" because "[this.in](http://this.in/)" is null
        at __randomizedtesting.SeedInfo.seed([F9345050587589D4]:0)
        at app//org.apache.lucene.index.ConcurrentMergeScheduler.handleMergeException(ConcurrentMergeScheduler.java:735)
        at app//org.apache.lucene.index.ConcurrentMergeScheduler$MergeThread.run(ConcurrentMergeScheduler.java:727)
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.lucene.index.PointValues.size()" because "[this.in](http://this.in/)" is null
        at org.apache.lucene.index.SortingCodecReader$SortingPointValues.size(SortingCodecReader.java:116)
        at org.apache.lucene.codecs.PointsWriter.mergeOneField(PointsWriter.java:51)
        at org.apache.lucene.codecs.PointsWriter.merge(PointsWriter.java:231)
        at org.apache.lucene.codecs.lucene90.Lucene90PointsWriter.merge(Lucene90PointsWriter.java:187)
        at org.apache.lucene.tests.codecs.asserting.AssertingPointsFormat$AssertingPointsWriter.merge(AssertingPointsFormat.java:133)
        at org.apache.lucene.index.SegmentMerger.mergePoints(SegmentMerger.java:185)
        at org.apache.lucene.index.SegmentMerger.mergeWithLogging(SegmentMerger.java:298)
        at org.apache.lucene.index.SegmentMerger.merge(SegmentMerger.java:145)
        at org.apache.lucene.index.IndexWriter.mergeMiddle(IndexWriter.java:5244)
        at org.apache.lucene.index.IndexWriter.merge(IndexWriter.java:4732)
        at org.apache.lucene.index.IndexWriter$IndexWriterMergeSource.merge(IndexWriter.java:6533)
        at org.apache.lucene.index.ConcurrentMergeScheduler.doMerge(ConcurrentMergeScheduler.java:639)
        at org.apache.lucene.index.ConcurrentMergeScheduler$MergeThread.run(ConcurrentMergeScheduler.java:700)

    Reproduce with: gradlew :lucene:core:test --tests "org.apache.lucene.search.TestPointQueries.testWrongNumBytes" -Ptests.jvms=4 -Ptests.jvmargs= -Ptests.seed=F9345050587589D4 -Ptests.multiplier=2 -Ptests.gui=true -Ptests.file.encoding=UTF-8 -Ptests.vectorsize=512
```

### Version and environment details

OpenJDK 11, 9.9.0 RC0.

--- a/lucene/core/src/java/org/apache/lucene/index/SlowCompositeCodecReaderWrapper.java
+++ b/lucene/core/src/java/org/apache/lucene/index/SlowCompositeCodecReaderWrapper.java
@@ -558,7 +558,7 @@ final class SlowCompositeCodecReaderWrapper extends CodecReader {
     private final int docBase;
 
     PointValuesSub(PointValues sub, int docBase) {
-      this.sub = sub;
+      this.sub = Objects.requireNonNull(sub);
       this.docBase = docBase;
     }
   }
@@ -599,7 +599,11 @@ final class SlowCompositeCodecReaderWrapper extends CodecReader {
         FieldInfo fi = codecReaders[i].getFieldInfos().fieldInfo(field);
         if (fi != null && fi.getPointDimensionCount() > 0) {
           PointValues v = readers[i].getValues(field);
-          values.add(new PointValuesSub(v, docStarts[i]));
+          if (v != null) {
+            // Apparently FieldInfo can claim a field has points, yet the returned
+            // PointValues is null
+            values.add(new PointValuesSub(v, docStarts[i]));
+          }
         }
       }
       if (values.isEmpty()) {

--- a/lucene/core/src/java/org/apache/lucene/index/SortingCodecReader.java
+++ b/lucene/core/src/java/org/apache/lucene/index/SortingCodecReader.java
@@ -24,6 +24,7 @@ import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.Objects;
 import org.apache.lucene.codecs.DocValuesProducer;
 import org.apache.lucene.codecs.FieldsProducer;
 import org.apache.lucene.codecs.KnnVectorsReader;
@@ -77,7 +78,7 @@ public final class SortingCodecReader extends FilterCodecReader {
     private final Sorter.DocMap docMap;
 
     SortingPointValues(final PointValues in, Sorter.DocMap docMap) {
-      this.in = in;
+      this.in = Objects.requireNonNull(in);
       this.docMap = docMap;
     }
 
@@ -472,6 +473,10 @@ public final class SortingCodecReader extends FilterCodecReader {
 
       @Override
       public PointValues getValues(String field) throws IOException {
+        var values = delegate.getValues(field);
+        if (values == null) {
+          return null;
+        }
         return new SortingPointValues(delegate.getValues(field), docMap);
       }
 

